scalding-core/src/main/scala/com/twitter/scalding/typed/Grouped.scala

```diff
@@ -77,15 +87,81 @@ object Grouped {
   def apply[K, V](pipe: TypedPipe[(K, V)])(implicit ordering: Ordering[K]): Grouped[K, V] =
     IdentityReduce(ordering, pipe, None)
 
-  def keySorting[T](ord: Ordering[T]): Fields = sorting("key", ord)
-  def valueSorting[T](implicit ord: Ordering[T]): Fields = sorting("value", ord)
+  def valueSorting[V](ord: Ordering[V]): Fields = Field.singleOrdered[V]("value")(ord)
 
-  def sorting[T](key: String, ord: Ordering[T]): Fields = {
-    val f = new Fields(key)
-    f.setComparator(key, ord)
-    f
+  /**
+   * If we are using OrderedComparable, we need to box the key
+   * to prevent other serializers from handling the key
+   */
+  def maybeBox[K, V](ord: Ordering[K])(op: (TupleSetter[(K, V)], Fields) => Pipe): Pipe = ord match {
+    case ordser: OrderedSerialization[K] =>
+      val (boxfn, cls) = Boxed.next[K]
```

Comment:
Is having 99 Box classes needed because something outside of our control (kryo?) requires a Class to be the key in some map?